service-front/module/Application/src/Controller/Authenticated/SessionKeepAliveController.php

```diff
@@ -12,4 +12,34 @@ public function indexAction()
         return new JsonModel(['refreshed' => $this->getSessionManager()->sessionExists()]);
     }
 
+    public function setExpiryAction()
+    {
+        if ($this->request->isPost()) {
+            // Derive expireInSeconds from request body
+            $expireInSeconds = null;
+
+            $content = $this->request->getContent();
+            if ($content !== '') {
+                $decodedContent = json_decode($content, true);
+                if (array_key_exists('expireInSeconds', $decodedContent)) {
+                    $expireInSeconds = $decodedContent['expireInSeconds'];
+                }
+            }
+
+            if ($expireInSeconds === null) {
+                $response = $this->getResponse();
+                $response->setStatusCode(400);
+                $response->setContent('Malformed request');
+                return $response;
+            }
+
+            $remainingSeconds = $this->getAuthenticationService()->setSessionExpiry(intval($expireInSeconds));
+            return new JsonModel(['remainingSeconds' => $remainingSeconds]);
+        }
+
+        $response = $this->getResponse();
+        $response->setStatusCode(405);
+        $response->setContent('Bad request method');
```

Comment:
405 should be `method not allowed` rather than bad request method (which is more like the 400 above) - which is more relevant to this type of api error. (might be being pedantic!)